include/ws_functions/pwg.users.php

```diff
@@ -621,4 +621,53 @@ function ws_users_setInfo($params, &$service)
     ));
 }
 
+/**
+ * API method
+ * Adds a favorite image for the current user
+ * @param mixed[] $params
+ */
+function ws_addFavorite($params, &$service)
+{
+  global $user;
+  if (is_a_guest())
+  {
+    return new PwgError(403, 'User must be logged in.');
+  }
+
+  $query = '
+INSERT INTO '.FAVORITES_TABLE.'
```

Comment:
Should this be 'insert ignore'? What happens if the favorite is already there?

Reply:
Probably should. I can make the adjustments, however I'm not exactly confident any of these PRs will ever make it to the project at this point.

Reply:
Yeah, things can be slow sometimes I guess. But it'll never be merged if it's not correct. :)